core/src/main/java/org/apache/calcite/sql/pretty/SqlPrettyWriter.java

```diff
@@ -360,7 +353,7 @@ public SqlPrettyWriter() {
 
   /** Creates a {@link SqlWriterConfig} with Calcite's SQL dialect. */
   public static SqlWriterConfig config() {
-    return CONFIG;
+    return SqlWriterConfig.of().withDialect(CalciteSqlDialect.DEFAULT);
```

Comment:
I guess this is a slight change of behavior as this is not a constant but a new instance each time?

Reply:
fixed.